Include <gtksourceview/gtksourcelanguage.h>, to obtain a declaration
of the gtk_source_language_get_name function.  Implicit function
declarations are not part of C99, and future compilers might reject
them by default.

Fixed differently upstream via:

commit 3415fc0a96bb71ac0f611f0737e839b5322a9174
Author: Corentin Noël <corentin@elementary.io>
Date:   Wed Apr 4 22:51:50 2018 +0100

    Use single header with GtkSource

diff --git a/libgda-ui/data-entries/plugins/libmain.c b/libgda-ui/data-entries/plugins/libmain.c
index a30c2db2c3258111..b4793f42006f00dd 100644
--- a/libgda-ui/data-entries/plugins/libmain.c
+++ b/libgda-ui/data-entries/plugins/libmain.c
@@ -48,6 +48,7 @@
   #include <gtksourceview/gtksourcebuffer.h>
   #include <gtksourceview/gtksourcestyleschememanager.h>
   #include <gtksourceview/gtksourcestylescheme.h>
+  #include <gtksourceview/gtksourcelanguage.h>
 #endif
 
 static GdauiDataEntry *plugin_entry_filesel_create_func (GdaDataHandler *handler, GType type, const gchar *options);
